src/v3/src/components/InputPassword/InputPassword.tsx

```diff
@@ -45,33 +50,61 @@ const InputPassword: UISchemaElementComponent<UISchemaElementComponentWithValida
   const focusRef = useAutoFocus<HTMLInputElement>(focus);
   const hasErrors = typeof errors !== 'undefined';
 
-  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+  const [showPassword, setShowPassword] = useState<boolean>(false);
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
     const changedVal = e.currentTarget.value;
     setTouched?.(true);
     onChangeHandler(changedVal);
     onValidateHandler?.(setErrors, changedVal);
   };
 
+  const handleClickShowPassword = () => {
+    setShowPassword(!showPassword);
+  };
+
+  const handleMouseDownPassword = (e: ClickEvent) => {
+    e.preventDefault();
+  };
+
+  const isConfirmPassword = name === 'confirmPassword';
+
   return (
     <Box>
-      <PasswordInput
-        label={label + (required ? ' *' : '')}
-        tooltipLabel={
-          (isHidden: boolean) => (isHidden ? getTranslation(translations, 'show') : getTranslation(translations, 'hide'))
-        }
-        value={value}
-        name={name}
+      <InputLabel htmlFor={useOid(name)}>{label + (required ? ' *' : '')}</InputLabel>
+      <OutlinedInput
         id={name}
+        name={name}
         error={hasErrors}
+        inputRef={focusRef}
         onChange={handleChange}
+        type={showPassword ? 'text' : 'password'}
+        value={value}
         disabled={loading}
         fullWidth
         inputProps={{
           'data-se': name,
           'aria-describedby': describedByIds,
           ...attributes,
         }}
-        ref={focusRef}
+        startAdornment={null}
+        endAdornment={(
+          <InputAdornment position="end">
+            <Tooltip title={showPassword ? getTranslation(translations, 'hide') : getTranslation(translations, 'show')}>
+              <IconButton
+                // TODO: OKTA-558040 request translation keys for aria labels
+                aria-label={isConfirmPassword ? 'Show re-enter password' : 'Show password'}
```

Comment:
Can you move this into the `src/transformer/i18n/transformInputPassword.ts` transformer. You can name the key anything you want, the i18nKey should be something that doesnt exist, and set the defaultValue to this English string. Then you can add a TODO there for us to add the proper translations.

Additionally, remember, this component is not only used by password fields, it is also used by `answer` fields, so having it always say `show ... password` wouldn't be appropriate. We will need a different translation (or maybe one that accepts a parameter and interpolates the string) for this label.

i.e. something like 

oie.secret.field.toggle.label = Show ${0}

Then when we use it we can do something like 

```
loc('oie.secret.field.toggle.label', 'login', ['password'])
loc('oie.secret.field.toggle.label', 'login', ['re-enter password'])
loc('oie.secret.field.toggle.label', 'login', ['answer'])
...
```

^ This detail can be added to the translation ticket.

Reply:
Very good point good catch